dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/user/UserResourceIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+
+    static HttpServletResponse response;
+    static HttpServletRequest request;
+    static UserResource resource;
+    static User user;
+    static Host host;
+    static User adminUser;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        resource = new UserResource();
+        adminUser = TestUserUtils.getAdminUser();
+        host = new SiteDataGen().nextPersisted();
+        user = TestUserUtils.getChrisPublisherUser(host);
+        response = new MockHttpResponse();
+
+        //Check if role has any layout, if is empty add one
+        if(APILocator.getLayoutAPI().loadLayoutsForUser(user).isEmpty()) {
+            APILocator.getRoleAPI()
+                    .addLayoutToRole(APILocator.getLayoutAPI().findAllLayouts().get(0),
+                            APILocator.getRoleAPI().getUserRole(user));
+        }
+        //Add permissions to the host
+        final Permission readPermissionsPermission = new Permission( host.getPermissionId(),
+                APILocator.getRoleAPI().getUserRole(user).getId(), PermissionAPI.PERMISSION_READ, true );
+        APILocator.getPermissionAPI().save(readPermissionsPermission,host,adminUser,false);
+        final Host hostTest = APILocator.getHostAPI().find(host.getIdentifier(), adminUser, false);//To check if the permissions were given
+        assertNotNull(hostTest);
+
+    }
+
+    private static HttpServletRequest mockRequest() {
+        final MockHeaderRequest request = new MockHeaderRequest(
+                new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request())
+                                .request())
+                        .request());
+
+        request.setHeader("Authorization",
+                "Basic " + new String(Base64.encode("admin@dotcms.com:admin".getBytes())));
+
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CURRENT_HOST,host);
+        request.getSession().setAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID,host);
+
+        return request;
+    }
+
+    private void loginAs() throws Exception {
+        final LoginAsForm loginAsForm = new LoginAsForm.Builder().userId(user.getUserId()).build();
+        request = mockRequest();
+        final Response resourceResponse = resource.loginAs(request,response,loginAsForm);
+        assertNotNull(resourceResponse);
+        assertEquals(resourceResponse.getEntity().toString(), Status.OK.getStatusCode(),resourceResponse.getStatus());
+        assertEquals(user.getUserId(),request.getSession().getAttribute(WebKeys.USER_ID));
+        assertNull(request.getSession().getAttribute(WebKeys.USER));
+        assertEquals(adminUser.getUserId(),request.getSession().getAttribute(WebKeys.PRINCIPAL_USER_ID));
+    }
+
+    @Test
+    public void test_loginAs_success() throws Exception{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280129)